crates/stages/src/pipeline/set.rs

```diff
@@ -6,6 +6,17 @@ use std::{
     fmt::{Debug, Formatter},
 };
 
+/// Stages that require state.
+const STATE_STAGES: [StageId; 7] = [
```

Comment:
i don't like this since we want to move these apis to a separate crate and this is specific to reth's default stages

can we instead add a `disable_stages(mut self, &[StageId])` & `disable_stages_if`? and move this array to the command itself